File: java/src/main/java/S001.java
public class S001 {
    public static long solve() {
        return multipleSum(3, 999) + multipleSum(5, 999) - multipleSum(15, 999);
    }

    private static long multipleSum(long multiple, long limit) {
        long count = limit / multiple;
        long max = count * multiple;
        return (max + multiple) * count / 2;
    }
}

File: cpp/S001.cpp
#include <iostream>

using namespace std;

static const int LIMIT = 999;

int multiple_sum(int multiple, int limit) {
	int nElements = limit / multiple;
	int top = multiple * nElements;
	return (top + multiple) * nElements / 2;
}

int main() {
	cout << multiple_sum(3, LIMIT) + multiple_sum(5, LIMIT) - multiple_sum(15, LIMIT) << endl;
	return 0;
}
